openedx/features/genplus_features/genplus_learning/signals.py

```diff
@@ -27,7 +30,39 @@ def _listen_for_course_publish(sender, course_key, **kwargs):
 @receiver(post_save, sender=ClassEnrollment)
 def add_program_enrollment(sender, instance, created, **kwargs):
     if created:
-        genplus_learning_tasks.enroll_class_students_to_program.delay(
-            gen_class_id=instance.gen_class.pk,
-            program_id=instance.program.pk,
+        genplus_learning_tasks.enroll_class_students_to_program.apply_async(
+            args=[instance.gen_class.pk, instance.program.pk],
+            countdown=PROGRAM_ENROLLMENT_COUNTDOWN,
         )
+
+
+@receiver(m2m_changed, sender=Program.units.through)
+def program_units_changed(sender, instance, action, **kwargs):
+    pk_set = kwargs.pop('pk_set', None)
+    if action == "post_add":
+        if isinstance(instance, Program) and instance.is_current:
+            program_class_ids = instance.class_enrollments.all().values_list('gen_class', flat=True)
+            program_unit_ids = [str(course_key) for course_key in pk_set]
+            for class_id in program_class_ids:
+                genplus_learning_tasks.enroll_class_students_to_program.apply_async(
```

Comment:
@hasnain-saeed can we add comments to these signals that what purpose they are serving